app/components/NoticeTypeCheckboxes.tsx

```diff
@@ -157,29 +159,161 @@ const NoticeTypes = {
   ],
 }
 
-interface NoticeTypeCheckboxeProps {
+type DataRow = {
+  topic: string
+  latestRate: number
+}
+
+type FormattedData = {
+  text: DataRow[]
+  binary: DataRow[]
+  voevent: DataRow[]
+}
+
+interface NoticeTypeCheckboxProps {
   defaultSelected?: string[]
+  selectedFormat?: 'binary' | 'text' | 'voevent'
+  showTotalSelected?: boolean
+}
+
+export async function loadTriggerRateData() {
+  return await getAverageTriggerRate()
 }
 
 export function NoticeTypeCheckboxes({
   defaultSelected,
-}: NoticeTypeCheckboxeProps) {
+  selectedFormat = 'text',
+  showTotalSelected = false,
+}: NoticeTypeCheckboxProps) {
+  const [metrics, setMetrics] = useState<FormattedData>()
+  const [userSelected, setUserSelected] = useState<{ [key: string]: boolean }>(
+    {}
+  )
+  const [selectedCounter, setSelectedCounter] = useState(0)
+  const [alertEstimate, setAlertEstimate] = useState(0)
+  useEffect(() => {
```

Comment:
The `useEffect()` hook should not be necessary any more because the trigger data need not be fetched asynchronously. Move this out of the hook so that the component can be fully pre-rendered.

Reply:
Can do, will be moved in next commit